app/controllers/application_controller.rb

```diff
@@ -163,6 +164,14 @@ def remember_cookie_sync
     end
   end
 
+  def forward_to_app_config_domain
+    return unless request.get? && # Let's only redirect get requests for this purpose.
+      request.host == ENV["APP_DOMAIN"] && # If the request equals the original set domain, e.g. forem-x.forem.cloud.
+      ENV["APP_DOMAIN"] != SiteConfig.app_domain # If the app domain config has now been set, let's go there instead.
+
+    redirect_to "#{request.protocol}#{SiteConfig.app_domain}#{request.fullpath}"
```

Comment:
**suggestion/question (non-blocking):** Does it make sense to use `URL.url(request.fullpath)` instead?

The only difference I see between this and `URL.url(request.fullpath)` is that if someone makes a GET request using `http://forem-x.forem.cloud` this will maintain the `http` and redirect to `http://real-domain.com`.

Using `URL.url(request.fullpath)` will redirect `http://forem-x.forem.cloud` to `https://real-domain.com` (or whatever protocol was specified for the deployment.

Reply:
Yeah this makes sense, thanks.